### PR TITLE
[13805] Use Case - Record Replay with Rosbag 

### PR DESCRIPTION
**Pull request by jsantiago-eProsima**
Signed-off-by: Javier Santiago <javiersantiago@eprosima.com>

### CODE REVIEW

File: docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
+.. include:: ../../../03-exports/aliases.include
+.. include:: ../../../03-exports/aliases-api.include
+.. include:: ../../../03-exports/roles.include
+
+.. _rosbag_capture:
+
+Rosbag2 Record and Replay
+==========================
+
+Rosbag2 is a ROS application that can be used to capture DDS messages and store them on an sqlite database. This allows
```
Comment (EduPonz) [inline docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst:10]: In general, please break line after each sentence period

File: docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+.. _rosbag_capture:
+
+Rosbag2 Record and Replay
+==========================
+
+Rosbag2 is a ROS application that can be used to capture DDS messages and store them on an sqlite database. This allows 
+inspecting and replaying said messages at a later time. 
+
+Rosbag2 interactions with a non-ROS2 Fast DDS application
```
Comment (EduPonz) [inline docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst:13]:
```suggestion
Rosbag2 interactions with a native Fast DDS application
```

File: docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst (hunk 1 above)
Comment (EduPonz) [inline docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst:10]:
```suggestion
Rosbag2 is a ROS 2 application that can be used to capture DDS messages and store them on an sqlite database. This allows 
```

File: docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+---------------------------------------------------------
+
+Using rosbag2 to capture traffic between ROS talkers and listeners is straightforward. Using it to record and replay
+messages sent by Fast DDS participants outside ROS2 ecosystem requires some modifications. 
+
+Prerequisites
+^^^^^^^^^^^^^
+Being a ROS2 application, to install rosbag requires to have the ROS2 repository for the ROS2 distribution of choice
```
Comment (EduPonz) [inline docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst:21]: In general
```suggestion
Being a ROS 2 application, to install rosbag requires to have the ROS 2 repository for the ROS 2 distribution of choice
```

File: docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+IDL files
+^^^^^^^^^
+IDL files used to generate types for the application need to have the types nested inside both the module name and an idl 
+module, in that order, for instance:
+
+.. code-block:: bash
+
+   module helloworld_type{
+     module idl {
```
Comment (EduPonz) [inline docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst:36]: I'd clearly state that the IDL module must be named `idl`
```suggestion
     module idl {
```

File: docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+Rosbag2 can then be installed following the instructions on their `Github repository
+<https://github.com/ros2/rosbag2>`_
+
+IDL files
+^^^^^^^^^
+IDL files used to generate types for the application need to have the types nested inside both the module name and an idl 
+module, in that order, for instance:
+
+.. code-block:: bash
```
Comment (EduPonz) [inline docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst:33]: Please format IDL
```suggestion
.. code-block:: idl
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
Section under work (notice being shown at the beginning of the section). I approve conditional to keep working into improving this section.

File: docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+   [INFO] [1644320308.422292205] [rosbag2_recorder]: All requested topics are subscribed. Stopping discovery...
+
+Rosbag will proceed to create a folder with an SQLite database inside. The path to this database file can be
+used to replay the recorded messages. Open a subscriber application for the same topic and run:
+
+.. code-block:: bash
+
+   export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+   source new_typesupport/install/setup.bash
```
[the remark this answers is not in the document]
Comment (jsantiago-eProsima) [reply docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst:168]: Not necessarily true. The user might want to replay the messages from a different terminal or at a later date.

**Pull request merged**